From d2fda3dbfdee09a6152c909f51c3f54da137a49e Mon Sep 17 00:00:00 2001
From: Anatoliy Lysenko <a.o.lysenko@gmail.com>
Date: Sun, 3 Apr 2011 14:38:47 +0300
Subject: [PATCH 2/2] Fix bug in collection_singular_ids on has many through association with conditions and includes,
 when condtions references tables from includes.
 This commit will revert fix from https://github.com/rails/rails/commit/3436fdfc12d58925e3d981e0afa61084ea34736c , but tests is ok.

---
 .../associations/collection_association.rb         |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/activerecord/lib/active_record/associations/collection_association.rb b/activerecord/lib/active_record/associations/collection_association.rb
index 9f4fc44..80f0537 100644
--- a/activerecord/lib/active_record/associations/collection_association.rb
+++ b/activerecord/lib/active_record/associations/collection_association.rb
@@ -57,7 +57,7 @@ module ActiveRecord
         else
           column  = "#{reflection.quoted_table_name}.#{reflection.association_primary_key}"
 
-          scoped.select(column).except(:includes).map! do |record|
+          scoped.select(column).map! do |record|
             record.send(reflection.association_primary_key)
           end
         end
-- 
1.7.0.4

